src/main/java/seedu/address/logic/commands/GroupCommand.java

```diff
@@ -0,0 +1,93 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Add tags to a group of contacts at the same time. Command stops if one of the names is invalid.
+ */
+public class GroupCommand extends Command {
+    public static final String COMMAND_WORD = "group";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds tags to all the selected contacts. "
+            + "Parameters: "
+            + PREFIX_TAG + "friends "
+            + PREFIX_NAME + "NAME1, NAME2, NAME3, NAME4 ... no limit to number";
+    private static final String MESSAGE_FAILURE = "Group command has failed";
+
+    private boolean isFirstAdded = true;
+    private String tag;
+    private List<String> nameList;
+
+
+    /**
+     * Constructs a group command based on the inputs.
+     */
+    public GroupCommand(String inputTag, String inputNameList) {
+        tag = inputTag;
+        nameList = Arrays.asList(inputNameList.split(", "));
+    }
+
+    /**
+     * Creates a string that indicates all the names that have been added with the tag.
+     * It also indicates if all the name have been added or if some names have failed.
+     * @param model is the main ModelManager class used by the current user
+     * @param result is the string that contains the final message to be sent
+     * @param targetName is the string name that is used to identify the target person to be edited
+     * @return the message string
+     */
+    private String addTagToContact(Model model, String result, String targetName) {
+        for (Person temp : model.getAddressBook().getPersonList()) {
+            if (temp.isSamePerson(Person.createTempFakePerson(targetName))) {
+                Set<Tag> newTags = new HashSet<>(temp.getTags());
```

Comment:
Is there a reason for using Set instead of HashSet? I think if you use HashSet you don't have to import Set.